src/Microsoft.AspNetCore.SignalR.Redis/RedisHubLifetimeManager.cs

```diff
@@ -52,51 +53,68 @@ public class RedisHubLifetimeManager<THub> : HubLifetimeManager<THub>, IDisposab
             _channels = new RedisChannels(typeof(THub).FullName);
             _protocol = new RedisProtocol(hubProtocolResolver.AllProtocols);
 
-            var writer = new LoggerTextWriter(logger);
             RedisLog.ConnectingToEndpoints(_logger, options.Value.Options.EndPoints, _serverName);
-            _redisServerConnection = _options.Connect(writer);
+        }
 
-            _redisServerConnection.ConnectionRestored += (_, e) =>
+        private async Task EnsureRedisServerConnection()
+        {
+            if (_redisServerConnection == null)
             {
-                // We use the subscription connection type
-                // Ignore messages from the interactive connection (avoids duplicates)
-                if (e.ConnectionType == ConnectionType.Interactive)
+                try
                 {
-                    return;
-                }
+                    await _connectionLock.WaitAsync();
```

Comment:
this goes outside the try

Reply:
We should be consistent, but we already have inconsistency :(.

https://github.com/aspnet/SignalR/blob/8a3516284e0019dcaaffbe695c88907494382868/src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs#L116-L118

https://github.com/aspnet/SignalR/blob/8a3516284e0019dcaaffbe695c88907494382868/src/Microsoft.AspNetCore.SignalR.Core/HubConnectionContext.cs#L174-L177

Reply:
It does actually matter - https://stackoverflow.com/questions/24139084/semaphoreslim-waitasync-before-after-try-block